OAK-11870: - allow passing fullGCGeneration parameter to RevisionsCommand

shodaaan opened the pull request:
- added fullGCGeneration parameter for RevisionsCommand revisions (MongoDB garbage collection) command

Commit:
OAK-10813 : using projection while listing/removing root temp nodes

Commit:
OAK-10813 : using projection while listing children of an node

Commit:
OAK-10823 : replaced db.collection.remove with db.collection.deleteOne/deleteMany to make it compatible with both mongo & mongosh

reschke committed:
OAK-10824: Release Oak 1.64.0 - Candidate Release Notes

ahanikel committed:
OAK-10771 - Add disk cache size stats and issue warning if evicted segment has zero length (#1427)

* OAK-10771 - Log if evicted segment has length zero.

* OAK-10771 - Add diskCacheSize stats

* OAK-10771 - Record cache size and change

---------

Co-authored-by: Axel Hanikel <ahanikel@adobe.com>

Commit:
OAK-10826 - Misleading warning about missing checkpoint

Commit:
OAK-10823 : reverted un-intended changes from var to let/const

reschke committed:
OAK-10824: Release Oak 1.64.0 - Candidate Release Notes

reschke committed:
[maven-release-plugin] prepare release jackrabbit-oak-1.64.0

reschke committed:
[maven-release-plugin] prepare for next development iteration

nfsantos committed:
OAK-10789 - Log paths used for inclusing/exclusion for Mongo regex filters in job summary (#1478)

fabriziofortino committed:
OAK-10811: (oak-search-elastic) reduce contention in IndexTracker (#1466)

* OAK-10811: (oak-search-elastic) reduce contention in IndexTracker

* OAK-10811: ElasticIndexInfoProvider uses node api to get :status info

* OAK-10811: rolled back unneeded change

t-rana committed:
Issues/OAK-10675 (#1385)

* OAK-10675: add service principal support in oak-blob-cloud-azure

* OAK-10675: use user delegation key signed sas for service principal and add license to new files

* OAK-10675: some self review changes

* OAK-10675: set null blob headers as empty string

* OAK-10675: update pom to resolve package dependency

* OAK-10675: add import packages to resolve bundle

* OAK-10770: embed runtime dependencies of azure identity in oak-segment-azure

* OAK-10770: remove unused imports, refactor dependencies

* OAK-10770: add imports, refactor dependencies

* OAK-10675: fix runtime dependency issues, code review

reschke committed:
OAK-10829: examples: update build-helper-maven-plugin to 3.6.0 (#1483)

mbaedke committed:
OAK-10624: o.a.j.o.namepath.JcrPathParser does not accept some valid local names containing '{' or '}'. (#1285)

Fixed issue and added test case.

Co-authored-by: Julian Reschke <reschke@apache.org>

mbaedke committed:
OAK-6756: Convert oak-auth-external to OSGi R7 annotations (#1371)

Done

---------

Co-authored-by: Julian Sedding <jsedding@apache.org>

t-rana committed:
OAK-10780: add azure access token refresh logic (#1441)

* OAK-10780: add azure access token refresh logic

* OAK-10780: modify log statement

* OAK-10780: modify log statement

* OAK-10780: reduce token refresh interval

reschke committed:
OAK-10830: Upgrade jackson-databind dependency to 2.17.1 (#1486)

reschke committed:
Revert "OAK-10830: Upgrade jackson-databind dependency to 2.17.1 (#1486)"

This reverts commit 4a4f2ed767da749f48df26bd1aea73ee262a02ed.

dulceanu committed:
OAK-10833 - Consistency check reports success for repository with SNFE in checkpoints (#1487)

reschke committed:
Revert "Revert "OAK-10830: Upgrade jackson-databind dependency to 2.17.1 (#1486)""

This reverts commit a9d0c1c81295dfde2e3b1f1bb97af589e5238816.

mbaedke committed:
OAK-10621: o.a.j.o.namepath.JcrPathParser does not accept indexed expanded names (#1488)

* OAK-10621: o.a.j.o.namepath.JcrPathParser does not accept indexed expanded names.

Fixed, added unit tests.

reschke committed:
OAK-10824: Release Oak 1.64.0

nfsantos committed:
OAK-10825 - Indexing job: log [TASK:<PHASE>:FAIL] log message for when an indexing phase fails. (#1481)

reschke committed:
OAK-10838: oak-lucene: maven install fails in oak-lucene due to baseline check (#1491)

mbaedke committed:
OAK-6761: Convert oak-blob-plugins to OSGi R7 annotations (#1415)

done

t-rana committed:
Issues/oak 10781 (#1485)

* OAK-10675: add service principal support in oak-blob-cloud-azure

* OAK-10675: use user delegation key signed sas for service principal and add license to new files

* OAK-10675: some self review changes

* OAK-10675: set null blob headers as empty string

* OAK-10675: update pom to resolve package dependency

* OAK-10675: add import packages to resolve bundle

* OAK-10770: embed runtime dependencies of azure identity in oak-segment-azure

* OAK-10770: remove unused imports, refactor dependencies

* OAK-10770: add imports, refactor dependencies

* OAK-10675: fix runtime dependency issues, code review

* OAK-10781: add access token refresh mechanism

* close executor

* reduce token refresh delauy to 1 minute

* retrigger build

rishabhdaim committed:
OAK-10739 : Provide support for FullGC in Mongo DocumentStore (#1454)

Co-authored-by: Stefan Egli <stefanegli@apache.org>
Co-authored-by: Rishabh Kumar <diam@adobe.com>
Co-authored-by: Jose Cordero <corderob@adobe.com>
Co-authored-by: Daniel Iancu <iancu@adobe.com>

Full details in branch: DetailedGC/OAK-10199

stefan-egli committed:
OAK-10843 : temporarily disable flaky tests

adelina-marian committed:
OAK-10837: Add documentation for UT/IT NodeStore fixtures

stefan-egli committed:
OAK-10844 : removing some test combinations and adding logging to help narrow down where time is spent

dulceanu committed:
OAK-10807 - Add oak-run diff to Azure repositories (#1496)

stefan-egli committed:
OAK-10843 : exclude a flaky test combination

stefan-egli committed:
OAK-10844 : marking a slow test as ignored for now

stefan-egli committed:
OAK-10846 : ignoring testPartialMergeRootCleanup as it seems to leak a thread

fabriziofortino committed:
OAK-10841: log error cause when ES cannot process a query (#1492)

reschke committed:
OAK-10847: query-spi: bump log level to ERROR for use of deprecated slf4j.event.Level method (#1500)

reschke committed:
OAK-10852: oak-store-document: disable unreliable tests in BranchCommitGCTest

stefan-egli committed:
OAK-10853 : temporarily disable flaky test

nfsantos committed:
OAK-10851 -  Log START/END/FAIL messages of indexing job phases inside the same package (#1501)

Joscorbe committed:
OAK-10854: Allow oak.findOne to find long path documents

stefan-egli committed:
OAK-10852 : refinement of 46476bcbfc to only ignore flaky mode

reschke committed:
OAK-10856: oak-run: TextPopulatorTest uses transitive dependency com.beust.jcommander.internal.Maps (#1505)

reschke committed:
OAK-10852: disable BranchCommitGCTest unmergedAddsThenMergedAddThenUnmergedRemovesChildren for FullGCMode.ORPHANS_EMPTYPROPS_UNMERGED_BC (#1506)

stefan-egli committed:
OAK-10852 : excluding another flaky test combination

reschke committed:
OAK-10860: oak-lucene: update gson test dependency and stop using deprecated APIs (#1509)

reschke committed:
OAK-10861: avoid use of Tika WriteOutHandler convenience method removed in Tika 2.x (#1510)

dulceanu committed:
OAK-9455 - Improve oak-run check to allow fast remote consistency checks (#1508)

nfsantos committed:
OAK-10868 - Use a constant to represent empty StringValue, to avoid allocating new objects in this case. (#1512)

nfsantos committed:
OAK-10866 - Avoid creating an intermediate String when appending a string value to a JsopBuilder. (#1511)

nfsantos committed:
OAK-10872 - Replace use of StringBuilder by string concatenation in PathUtils.concat(String, String)

stefan-egli committed:
OAK-10870 : New oak-mongo.js methods for sizes of properties, counts … (#1514)

* OAK-10870 : New oak-mongo.js methods for sizes of properties, counts by clusterId and unmerged branch commits

* OAK-10870 : addressed review comments

ionutzpi committed:
OAK-10803 - Modified DocumentPropertyState.java to compress values ba… (#1473)

Co-authored-by: pirlogea <pirlogea@adobe.com>

nscendoni committed:
OAK-10451 UserPrincipalProvider may cause many conflicts when under load (#1128)

Co-authored-by: anchela <angela@apache.org>
Co-authored-by: Marcel Reutegger <marcel.reutegger@gmail.com>

reschke committed:
OAK-10873: store-spi: remove use of Guava CharMatcher in test case (#1516)

* OAK-10873: store-spi: remove use of Guava CharMatcher in test case

* OAK-10873: store-spi: remove use of Guava CharMatcher in test case - use less expensive detection

* OAK-10873: store-spi: remove use of Guava CharMatcher in test case - off by one

* Revert "OAK-10873: store-spi: remove use of Guava CharMatcher in test case - off by one"

This reverts commit d78fe0af400f23ac4f1885b5cca6635981ebb94b.

anchela committed:
OAK-10451 : UserPrincipalProvider may cause many conflicts when under load (add missing licence header)

nfsantos committed:
OAK-10879 - Indexing job: In the index report, list phase timings in chronological order and not alphabetic order (#1519)

reschke committed:
Revert "OAK-10803 - Modified DocumentPropertyState.java to compress values ba… (#1473)"

This reverts commit 163352368343164671231b1ee8f4db75f3925877.

reschke committed:
OAK-10883: document mk: disable flaky test VersionGarbageCollectorIT.testBundledPropUnmergedBCGC for mode ORPHANS_EMPTYPROPS_UNMERGED_BC (#1525)

shodaaan committed:
- create class FullGCOptions which will hold the different parameters used for running full GC
- added variables to Configuration.java for 2 full GC modes: GAP_ORPHANS and EMPTY_PROPERTIES
- udpated VersionGarbageCollector constructor, as well as DocumentNodeStore Builder / Service and DocumentNodeStore to get values received from Configuration via the FullGCOptions class

shodaaan committed:
Merge remote-tracking branch 'origin/trunk' into trunk

# Conflicts:
#	RELEASE-NOTES.txt
#	oak-api/pom.xml
#	oak-auth-external/pom.xml
#	oak-auth-ldap/pom.xml
#	oak-authorization-cug/pom.xml
#	oak-authorization-principalbased/pom.xml
#	oak-benchmarks-elastic/pom.xml
#	oak-benchmarks-lucene/pom.xml
#	oak-benchmarks-solr/pom.xml
#	oak-benchmarks/pom.xml
#	oak-blob-cloud-azure/pom.xml
#	oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobContainerProvider.java
#	oak-blob-cloud-azure/src/test/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureDataStoreUtils.java
#	oak-blob-cloud/pom.xml
#	oak-blob-plugins/pom.xml
#	oak-blob/pom.xml
#	oak-commons/pom.xml
#	oak-core-spi/pom.xml
#	oak-core/pom.xml
#	oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/ConsoleIndexingReporter.java
#	oak-core/src/main/java/org/apache/jackrabbit/oak/security/user/UserPrincipalProvider.java
#	oak-core/src/test/java/org/apache/jackrabbit/oak/security/user/CachedPrincipalMembershipReaderTest.java
#	oak-doc-railroad-macro/pom.xml
#	oak-doc/pom.xml
#	oak-examples/pom.xml
#	oak-examples/standalone/pom.xml
#	oak-examples/webapp/pom.xml
#	oak-exercise/pom.xml
#	oak-http/pom.xml
#	oak-it-osgi/pom.xml
#	oak-it/pom.xml
#	oak-jackrabbit-api/pom.xml
#	oak-jcr/pom.xml
#	oak-lucene/pom.xml
#	oak-parent/pom.xml
#	oak-pojosr/pom.xml
#	oak-query-spi/pom.xml
#	oak-run-commons/pom.xml
#	oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/pipelined/PipelinedMongoDownloadTask.java
#	oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/pipelined/PipelinedTransformTask.java
#	oak-run-commons/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStoreHelper.java
#	oak-run-elastic/pom.xml
#	oak-run/pom.xml
#	oak-run/src/main/java/org/apache/jackrabbit/oak/run/RevisionsCommand.java
#	oak-run/src/test/java/org/apache/jackrabbit/oak/plugins/document/RevisionsCommandTest.java
#	oak-run/src/test/java/org/apache/jackrabbit/oak/plugins/tika/TextPopulatorTest.java
#	oak-search-elastic/pom.xml
#	oak-search/pom.xml
#	oak-security-spi/pom.xml
#	oak-segment-aws/pom.xml
#	oak-segment-azure/pom.xml
#	oak-segment-azure/src/main/java/org/apache/jackrabbit/oak/segment/azure/tool/AzureCheck.java
#	oak-segment-azure/src/main/java/org/apache/jackrabbit/oak/segment/azure/tool/ToolUtils.java
#	oak-segment-remote/pom.xml
#	oak-segment-tar/pom.xml
#	oak-shaded-guava/pom.xml
#	oak-solr-core/pom.xml
#	oak-solr-osgi/pom.xml
#	oak-store-composite/pom.xml
#	oak-store-document/pom.xml
#	oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java
#	oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStore.java
#	oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStoreBuilder.java
#	oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStoreService.java
#	oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGCRecommendations.java
#	oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGCSupport.java
#	oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java
#	oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoVersionGCSupport.java
#	oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/rdb/RDBDocumentNodeStoreBuilder.java
#	oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/util/Utils.java
#	oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/BranchCommitGCTest.java
#	oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStoreServiceConfigurationTest.java
#	oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/VersionGCSupportTest.java
#	oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/VersionGCTest.java
#	oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollectorIT.java
#	oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoDocumentNodeStoreBuilderTest.java
#	oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/rdb/RDBDocumentNodeStoreBuilderTest.java
#	oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/util/UtilsTest.java
#	oak-store-spi/pom.xml
#	oak-upgrade/pom.xml
#	pom.xml

shodaaan committed:
Merge remote-tracking branch 'origin/trunk' into trunk

shodaaan committed:
Merge remote-tracking branch 'origin/trunk' into trunk

shodaaan committed:
Merge remote-tracking branch 'origin/trunk' into trunk

shodaaan committed:
Merge remote-tracking branch 'origin/trunk' into trunk

shodaaan committed:
Merge remote-tracking branch 'origin/trunk' into trunk

shodaaan committed:
Merge branch 'refs/heads/trunk' into trunk_Horia

# Conflicts:
#	oak-run-commons/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStoreHelper.java
#	oak-run/src/main/java/org/apache/jackrabbit/oak/run/RevisionsCommand.java
#	oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java
#	oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStore.java
#	oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStoreBuilder.java
#	oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStoreService.java
#	oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java

shodaaan committed:
Merge branch 'refs/heads/trunk' into trunk_Horia

shodaaan committed:
Merge branch 'refs/heads/trunk' into trunk_Horia

shodaaan committed:
Merge branch 'trunk' into trunk_Horia

shodaaan committed:
Merge branch 'apache:trunk' into trunk

shodaaan committed:
Merge remote-tracking branch 'Oak_Jackrabbit_Horia/trunk' into trunk_Horia

shodaaan committed:
Merge branch 'trunk' of https://github.com/shodaaan/jackrabbit-oak into trunk_Horia

shodaaan committed:
Merge branch 'trunk' into trunk_Horia

shodaaan committed:
Merge branch 'trunk' of https://github.com/shodaaan/jackrabbit-oak into trunk_Horia

shodaaan committed:
Merge branch 'trunk' into trunk_Horia

# Conflicts:
#	oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/util/Utils.java

shodaaan committed:
Merge branch 'trunk' into trunk_Horia

shodaaan committed:
Merge branch 'trunk' into trunk_Horia

shodaaan committed:
- added functionality for passing fullGCGeneration variable to RevisionsCommand (oak-run)

shodaaan committed:
- removed old file FullGCOptions

Joscorbe reviewed (approved):
It looks good, but please add a test case that covers the new parameter [with the default value here](https://github.com/apache/jackrabbit-oak/blob/trunk/oak-run/src/test/java/org/apache/jackrabbit/oak/plugins/document/RevisionsCommandTest.java#L190).

And a new test case that covers the non-default value, [like this](https://github.com/apache/jackrabbit-oak/blob/trunk/oak-run/src/test/java/org/apache/jackrabbit/oak/plugins/document/RevisionsCommandTest.java#L376-L391).

Otherwise the coverage checks on new code will fail the pipeline.

rishabhdaim commented on oak-run/src/main/java/org/apache/jackrabbit/oak/run/RevisionsCommand.java:
```diff
@@ -429,6 +437,7 @@ private VersionGarbageCollector bootstrapVGC(RevisionsOptions options, Closer cl
         System.out.println("IncludePaths are : " + sortedSet(gc.getFullGCIncludePaths()));
         System.out.println("ExcludePaths are : " + sortedSet(gc.getFullGCExcludePaths()));
         System.out.println("FullGcMode is : " + VersionGarbageCollector.getFullGcMode());
+        System.out.println("FullGCGeneration is : " + options.getFullGcGeneration());
```
```suggestion
        System.out.println("FullGCGeneration is : " + gc.getFullGcGeneration());
```

rishabhdaim replied:
To read values from `VersionGarbageCollector`

shodaaan replied:
Done, thank you for the suggestion.

rishabhdaim reviewed (changes requested):
Please add test cases for fullGcGeneration as well.

shodaaan committed:
- added getter for fullGCGeneration to VersionGarbageCollector and updated RevisionsCommand output to use this getter when logging start
- added unit tests for default and set value of fullGCGeneration in RevisionsCommand

shodaaan commented:
Added unit tests.

Joscorbe commented:
Ok, build passed but the sonarqube checks failed because it doesn't have permissions on the fork.

I will merge it then.
